web-locks/resources/request-around-navigation.html

```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+<head><meta charset="utf-8"></head>
+<body>
+<script>
+'use strict';
+(async () => {
+  const [uuid, phase] = location.search.split('_');
+  const neverSettled = new Promise(() => {});
+  navigator.locks.request(uuid, () => neverSettled);
```

Comment:
A more robust version:

```js
let firstRequestGranted = false;
navigator.locks.request(lockName, () => {
  firstRequestGranted = true;
  return neverSettled;
}); 
let secondRequestGranted = false;
navigator.locks.request(lockName, () => {
  secondRequestGranted = true;
  return neverSettled;
}); 
```

This doesn't depend on `navigator.locks.query`, which may have its own bugs. The frame would postMessage `{firstRequestGranted, secondRequestGranted}`.

Reply:
Hm, I spoke too soon. I think that the test would be more robust if we waited for the first request to be granted, then requested an unrelated lock and waited for that to be granted. 

```js
let firstRequestPromiseResolver;
const firstRequestPromise =
    new Promise(resolve => { firstRequestPromiseResolver = resolve; });
navigator.locks.request(lockName, () => {
  firstRequestPromiseResolver();
  return neverSettled;
});
await firstRequestPromise;
// This request should never get granted.
let secondRequestGranted = false;
navigator.locks.request(lockName, () => {
  secondRequestGranted = true;
  return neverSettled;
});
let lastRequestPromiseResolver;
// Give the lock manager a chance to grant locks.
const lastRequestPromise =
    new Promise(resolve => { lastRequestPromiseResolver = resolve; });
navigator.locks.request(token(), () => {
  lastRequestPromiseResolver();
  return neverSettled;
}); 
await lastRequestPromise;

self.parent.postMessage({ secondRequestGranted, state });
```